lib/devise/models/confirmable.rb

```diff
@@ -169,6 +169,7 @@ def skip_reconfirmation!
         # in models to map to a nice sign up e-mail.
         def send_on_create_confirmation_instructions
           send_confirmation_instructions
+          skip_reconfirmation!
```

Comment:
@ulissesalmeida  I'll try. What about this PR at all? Does anybody else faced same problem? After this PR updating record right after creation works different:
```ruby
user = User.create(email: some_mail)
user.update(email: new_email) # this will NOT send confirmation instructions
user.reload.update(email: new_email) # this too
User.find(user.id).update(email: new_email) # but this will
```
IMO This is not consistent behavior, PR solves some app-specific case and should not be part of Devise. WDYT?